common/dynamicconfig/constants.go

```diff
@@ -745,6 +745,12 @@ const (
 	MaximumSignalsPerExecution = "history.maximumSignalsPerExecution"
 	// ShardUpdateMinInterval is the minimal time interval which the shard info can be updated
 	ShardUpdateMinInterval = "history.shardUpdateMinInterval"
+	// ShardUpdateMinTasksCompleted is the minimum number of tasks which must be completed (across all queues) before the shard info can be updated.
+	// Note that once history.shardUpdateMinInterval amount of time has passed we'll update the shard info regardless of the number of tasks completed.
+	// When the this config is zero or lower we will only update shard info at most once every history.shardUpdateMinInterval.
+	ShardUpdateMinTasksCompleted = "history.shardUpdateMinTasksCompleted"
+	// ShardUpdateQueueMetricsInterval is the minimum amount of time between updates to a shard's queue metrics
+	ShardUpdateQueueMetricsInterval = "history.shardUpdateQueueMetricsInterval"
```

Comment:
nit: I am also fine with hardcoding this value. Do you have any case in mind where the internal needs to be tuned?

Reply:
No. It's a habit